### ISSUE TITLE
Tasks as Subtask

### ISSUE THREAD

**Issue by aegis301** [helpwave/helpwave#56]
### Describe the feature

Tasks should be able to be placed as subtasks for other tasks like in Todoist.
![Screenshot 2023-08-14 at 15 55 00](https://github.com/helpwave/helpwave/assets/57251443/afb31974-2842-4d8b-9121-7e7f1fa874a4)

### Why is this feature needed?

We need this feature to build complet pipelines of tasks and subtasks and to build templates of care tracks. E.g. If a patient gets a fever, several tasks are implied. E.g. Blood cultures, Chest-X-Ray etc. Chest-X-Ray however can also be used as a template outside of the context of rising temperature and includes several other subtasks.

**Comment by use-to** [comment]
Isn't it subtasks as tasks?

**Comment by DasProffi** [comment]
Additional information for the implementation:
- Task and subtasks are hierarchical, but may also be linked for directly accessing a subtask
- Only subtasks of depth 2 or less are needed to be shown in menus
- Completing a task will complete all subtasks
- Assignment to tasks will result in being assigned to all subtasks
- If currently a Task lower in the hierarchy is shown primarily, the root task of the hierarchy can or can't be visible

The picture below shows a possible hierarchy, "Task 1" would be the root Task
![Task-Hierarchy (2)](https://github.com/helpwave/helpwave/assets/67233923/656b26aa-d9c7-44dc-8e17-13e01063de30)

**Comment by aegis301** [comment]
Quick comment on point 3. Are we sure we want to do that? I could imagine a situation where team members want to split up responsibility for a subtask. 

Possible Scenario: We have to make a diagnostic workup of unclear fever at the ward. So I will call radiology in order to get a chest x-ray while my colleague will go ahead and draw a blood sample.

**Comment by sophiafe** [comment]
@aegis301, point 4 might have to be elaborated a bit further. Felix and I discussed that it would be good if the default responsibility when adding a new subtask would be the same person that is also responsible for the main task. However, there should be the possibility to manually change the responsibility of the subtasks or create templates in which the responsibility of the subtasks are already assigned to several people. Does this make sense to you?

**Comment by aegis301** [comment]
Yeah sure, good idea so team member can reassign responsibilities manually. Perfect! However, I think if I de-assign myself from a task, this should also trickle down to other tasks. So e.g. I just quickly jotted in a few tasks for a patient without intention to actually do them.

**Comment by sophiafe** [comment]
That's true. However, there should never be a task that is not assigned to anybody right? So maybe we would need a function for "nursing" or "medical" in general so that the team can then do the assignment?

**Comment by aegis301** [comment]
Why shouldn't there be? IMO it is completely realistically that there are unassigned tasks. I imagine rounds on the ward and during these rounds tasks will just quickly be jotted into the app. After the attending left, the residents and nurses will then distribute tasks among each other and use it to communicate their status and work. Or do you have another process in mind?

**Comment by sophiafe** [comment]
I totally agree. However, if the resident jotting down the task will be marked as responsible by default, he will also feel responsible for all tasks to be distributed to the correct people. Imo it would just be a sort of quality management but I would also understand it if teams would prefer for those tasks to not be assigned in the beginning
